### ISSUE TITLE
Conversation.SendAsync not returning in default application

### ISSUE THREAD

**Issue by GiuseppePiscopo** [Microsoft/BotBuilder#4568]
## Bot Info
* SDK Platform: .NET
* SDK Version: 3.15.2
* Active Channels: none, local emulation
* Deployment Environment: localhost Bot Framework Emulator 3.5.36

## Issue Description
Working with default project, as generated from Bot Application VS template, and *after* having updated its dependencies, execution reaches `await Conversation.SendAsync(...` in `MessagesController.Post` and never returns from there. Probably it times out. Anyway, `RootDialog` is never instantiated.  

I tried taking the generated project **as-is**, without updating dependencies, and the issue disappears.

See also [this SO thread](https://stackoverflow.com/questions/50195115/conversation-sendasync-not-returning-in-bot-framework-default-application) for reference.

## Code Example
It's the one generated from Bot Application Visual Studio template: 

```c#
// MessagesController.cs

[BotAuthentication]
public class MessagesController : ApiController
{
    /// <summary>
    /// POST: api/Messages
    /// Receive a message from a user and reply to it
    /// </summary>
    public async Task<HttpResponseMessage> Post([FromBody]Activity activity)
    {
        if (activity.Type == ActivityTypes.Message)
        {
            await Conversation.SendAsync(activity, () => new Dialogs.RootDialog());
        }
        else
        {
            HandleSystemMessage(activity);
        }
        var response = Request.CreateResponse(HttpStatusCode.OK);
        return response;
    }

    private Activity HandleSystemMessage(Activity message)
    // ...
}

// RootDialog.cs

[Serializable]
public class RootDialog : IDialog<object>
{
    public Task StartAsync(IDialogContext context)
    {
        context.Wait(MessageReceivedAsync);

        return Task.CompletedTask;
    }

    private async Task MessageReceivedAsync(IDialogContext context, IAwaitable<object> result)
    // ...
}
```

Here's the `packages.config`:

```xml
<?xml version="1.0" encoding="utf-8"?>
<packages>
  <package id="Autofac" version="3.5.2" targetFramework="net46" />
  <package id="Chronic.Signed" version="0.3.2" targetFramework="net46" />
  <package id="Microsoft.AspNet.WebApi" version="5.2.3" targetFramework="net46" />
  <package id="Microsoft.AspNet.WebApi.Client" version="5.2.3" targetFramework="net46" />
  <package id="Microsoft.AspNet.WebApi.Core" version="5.2.3" targetFramework="net46" />
  <package id="Microsoft.AspNet.WebApi.WebHost" version="5.2.3" targetFramework="net46" />
  <package id="Microsoft.Bot.Builder" version="3.15.2" targetFramework="net46" />
  <package id="Microsoft.Bot.Connector" version="3.15.2" targetFramework="net46" />
  <package id="Microsoft.IdentityModel.Logging" version="1.1.4" targetFramework="net46" />
  <package id="Microsoft.IdentityModel.Protocol.Extensions" version="1.0.4.403061554" targetFramework="net46" />
  <package id="Microsoft.IdentityModel.Protocols" version="2.1.4" targetFramework="net46" />
  <package id="Microsoft.IdentityModel.Protocols.OpenIdConnect" version="2.1.4" targetFramework="net46" />
  <package id="Microsoft.IdentityModel.Tokens" version="5.1.4" targetFramework="net46" />
  <package id="Microsoft.Rest.ClientRuntime" version="2.3.2" targetFramework="net46" />
  <package id="Microsoft.WindowsAzure.ConfigurationManager" version="3.1.0" targetFramework="net46" />
  <package id="Newtonsoft.Json" version="9.0.1" targetFramework="net46" />
  <package id="System.IdentityModel.Tokens.Jwt" version="5.1.4" targetFramework="net46" />
</packages>
```

## Reproduction Steps
1. Follow directions from quickstart [Create a bot with the Bot Builder SDK for .NET](https://docs.microsoft.com/en-us/azure/bot-service/dotnet/bot-builder-dotnet-quickstart?view=azure-bot-service-3.0) to download VS templates, create project, update dependencies
1. Follow same directions to start project in VS, start emulator, connect emulator to bot
1. In `MessagesController.Post` set breakpoints on `await Conversation.SendAsync` and `var response = Request.CreateResponse` lines. 
1. In `RootDialog.StartAsync` set a breakpoint anywhere
1. From bot emulator send a text message 
1. Hit breakpoint on `await Conversation.SendAsync` 
1. Continue execution

## Expected Behavior
Breakpoint on `RootDialog` and `var response = Request.CreateResponse` should be hit. Emulator should show bot reply.

## Actual Results
None of those two mentioned breakpoints are hit. Emulator shows *couldn't send, retry* message.

**Comment by GiuseppePiscopo** [comment]
BTW tried with the same SDK version found in *BotBuilder-Samples\core-MultiDialogs* as of yesterday ([this commit](https://github.com/Microsoft/BotBuilder-Samples/commit/6c3d09c92ebeaf12ee7597a11bb09ac6e2fca6e5)):

```xml
<package id="Microsoft.Bot.Builder" version="3.12.2.4" targetFramework="net46" />
<package id="Microsoft.Bot.Connector" version="3.12.2.4" targetFramework="net46" />
```

and it works too.

**Comment by GiuseppePiscopo** [comment]
It looks like this has been fixed in [3.15.2.1](https://github.com/Microsoft/BotBuilder/releases/tag/botbuilder%403.15.2.1), just released. Thanks to the team!
